### ISSUE TITLE
Missing paths p-value different from expected

### ISSUE THREAD

**Issue by aolney** [jslefche/piecewiseSEM#145]
I'm doing a mediation analysis. My model looks like this:

```
glmer( b ~ a + ( 1 | id ) )
lmer( c ~ b + ( 1 | id ) )
```

I would include the direct effect `lmer( c ~ a + ( 1 | id ) )` but if I do, I get the error `"Collapse multiple single regressions into a single multiple regression so that each response appears only once!"` If I collapse as directed, i.e. lmer( c ~ b + a + ( 1 | id ) ), I get the error `All endogenous variables are conditionally dependent.`

When I run `sem.fit` on the indirect model, it rightly responds that there is a missing path and it is the direct effect. It gives an estimate and a p-value. However, if I run the direct effect model separately with `lmer` I get a different p-value and estimate:

- missing paths: estimate = -.0972 ; p-value = .0119
- lmer (lmerTest I think, with KR): estimate = -.0806 ; p-value = .0261

I'm wondering which estimate to trust more. Is the missing path estimate conditional on the model of the indirect effect?

**Comment by jslefche** [comment]
Hi Andrew,

Correct, specifying the additional path would require a multiple regression rather than two separate simple linear regressions, because the tests are of conditional independence (e.g., a on c given b).

If you have a fully saturated model, then you won’t have any tests of directed separation because all paths are included.
Its unclear how you are obtaining a P=value using `lmer` as it won’t return one by default. Is it possible you have `lmerTest` or some other package loaded? The approximation of the ddf are different and hence the different P-values.

J

____________________________
Jonathan S. Lefcheck, Ph.D.
Post-doctoral Researcher
Bigelow Laboratory for Ocean Science
Phone: +1 (207) 315-2567 ext. 420
Website: www.jonlefcheck.net

________________________________

**Comment by aolney** [comment]
Many thanks for those thoughts. If I specify `lme4::lmer` I get the same missing path coefficients as before, `piecewiseSEM` = -.0972 but `lme4::lmer`  = -.0806

Re the saturated model, it sounds like my problem was with calling `sem.fit` with a saturated model. So if I skip that call, build the following saturated model,
```
glmer( b ~ a + ( 1 | id ) )
lmer( c ~ a + b + ( 1 | id ) )
```
and then call `sem.coefs`, I get a new error:
```
Error in KRmodcomp.lmerMod(i, i.reduced) : 
  Models have either equal fixed mean stucture or are not nested
```
which [seems to be located in code here](https://github.com/jslefche/piecewiseSEM/blob/7d14d138b54c9ad53b4fed475436f6192b418932/R/sem.coefs.R#L84)

Any insight into that error would be appreciated.

**Comment by aolney** [comment]
@jslefche OK I now understand my error with the p-values! The output is truncated for a the missing path, appearing as `c ~ b + ...`. If I fully saturate, then separately run a regression for `c ~ a + b`, all p-values and coefficients match exactly. I did this all with `lm` so that I could compare to `lavaan` (now exactly matches that as well).

However, when I try with `glmer/lmer`, I still get the `KRmodcomp` error I got before.  

I believe I've also figured out that error, but I don't know how to correct it. When my `lmer` model hits this code in `sem.coef`
```
kr.p = sapply(names(fixef(i))[irow], function(x) {
          
          i.reduced = update(i, as.formula(paste("~ . -", x)))
        
          KRmodcomp(i, i.reduced)$test$p.value[1]
          
        } )
```
one of the fixed effects is a **level of a factor, rather than the factor itself**. As a result, the `update` call does not remove the variable, and the `KRmodcomp` call compares `c ~ a + b` with itself.

I'm going to start a new issue for this.
